src/transformers/models/blip_2/configuration_blip_2.py

```diff
@@ -306,6 +271,10 @@ class Blip2Config(PretrainedConfig):
     ```"""
 
     model_type = "blip-2"
+    sub_configs = ["text_config", "qformer_config", "vision_config"]
+    text_config_class = "AutoConfig"
+    qformer_config_class = "Blip2QFormerConfig"
+    vision_config_class = "Blip2VisionConfig"
```

Comment:
I think the subconfigs should be a dict: `{ "text_config": AutoConfig, "qformer_config": Blip2QFormerConfig, ...}` 
this way the init loop is "simpler", you already have the class, you just use it (and use the CONFIG_MAPPING) for AutoConfig

Reply:
On top of that, it scales and makes more sense imo!